src/protagonist/Engine/Ingest/Image/ImageServer/Clients/CantaloupeThumbsClient.cs

```diff
@@ -46,35 +46,47 @@ public class CantaloupeThumbsClient : IThumbsClient
         foreach (var size in thumbSizes)
         {
             ++count;
-            using var response =
-                await cantaloupeClient.GetAsync(
-                    $"iiif/3/{convertedS3Location}/full/{size}/0/default.jpg", cancellationToken);
+            var imageOnDisk = await GenerateSingleThumbnail(thumbFolder, convertedS3Location, size, assetId, count,
+                thumbsResponse, imageSize, true, cancellationToken);
             
-            if (response.StatusCode == HttpStatusCode.BadRequest)
+            if (imageOnDisk is not null)
             {
-                // This is likely an error for the individual thumb size, so just continue
-                await LogErrorResponse(response, assetId, size, LogLevel.Information, cancellationToken);
-                continue;
-            }
-            
-            if (response.IsSuccessStatusCode)
-            {
-                var imageOnDisk = await SaveImageToDisk(response, size, thumbFolder, count, cancellationToken);
                 thumbsResponse.Add(imageOnDisk);
                 ValidateSize(size, imageSize, imageOnDisk, assetId);
             }
-            else
-            {
-                await LogErrorResponse(response, assetId, size, LogLevel.Error, cancellationToken);
-                throw new HttpException(response.StatusCode, "failed to retrieve data from the thumbs processor");
-            }
         }
         
         return thumbsResponse;
     }
 
+    private async Task<ImageOnDisk?> GenerateSingleThumbnail(string thumbFolder,
+        string convertedS3Location, string size, AssetId assetId, int count, List<ImageOnDisk> thumbsResponse,
+        Size imageSize, bool retry, CancellationToken cancellationToken)
```

Comment:
Is it worth renaming `retry`? I initially took `retry` to mean this _is_ the retry, rather than it _should_ retry.